pkg/build/poller.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2024 Daytona Platforms Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package build
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/daytonaio/daytona/pkg/logs"
+	"github.com/daytonaio/daytona/pkg/poller"
+	"github.com/daytonaio/daytona/pkg/scheduler"
+	"github.com/daytonaio/daytona/pkg/telemetry"
+	log "github.com/sirupsen/logrus"
+)
+
+type PollerConfig struct {
+	Interval         string
+	Scheduler        scheduler.IScheduler
+	BuildPollerId    string
+	BuildStore       Store
+	BuilderFactory   IBuilderFactory
+	LoggerFactory    logs.LoggerFactory
+	TelemetryService telemetry.TelemetryService
+}
+
+type BuildPoller struct {
```

Comment:
Please rename this to `BuildRunner`. It's a bit more appropriate since it actually runs builds.